packages/main/src/plugin/kubernetes-context-state.ts

```diff
@@ -0,0 +1,243 @@
+      console.error(`==> ${context.name}: connect`, err);
+      const previous = this.contextsState.get(context.name);
+      if (previous) {
+        previous.reachable = err === undefined;
+      }
+      this.dispatchContextsState();
+    });
+    this.restartInformer(informer, context);
+    return informer;
+  }
+
+  private createDeploymentInformer(
+    kc: KubeConfig,
+    ns: string,
+    context: KubeContext,
+  ): Informer<V1Deployment> & ObjectCache<V1Deployment> {
+    const k8sApi = kc.makeApiClient(AppsV1Api);
+    const listFn = () => k8sApi.listNamespacedDeployment(ns);
+    const informer = makeInformer(kc, `/apis/apps/v1/namespaces/${ns}/deployments`, listFn);
+
+    informer.on('add', (_obj: V1Deployment) => {
+      const previous = this.contextsState.get(context.name);
+      if (previous) {
+        previous.deploymentsCount++;
+      }
+      this.dispatchContextsState();
+    });
+
+    informer.on('delete', (_obj: V1Deployment) => {
+      const previous = this.contextsState.get(context.name);
+      if (previous) {
+        previous.deploymentsCount--;
+      }
+      this.dispatchContextsState();
+    });
+    informer.on('error', (_err: unknown) => {
+      // Restart informer after 5sec
+      setTimeout(() => {
+        this.restartInformer(informer, context);
+      }, 5000);
+    });
+    this.restartInformer(informer, context);
+    return informer;
+  }
+
+  private createReplicasetInformer(
+    kc: KubeConfig,
+    ns: string,
+    context: KubeContext,
+  ): Informer<V1ReplicaSet> & ObjectCache<V1ReplicaSet> {
+    const k8sApi = kc.makeApiClient(AppsV1Api);
+    const listFn = () => k8sApi.listNamespacedReplicaSet(ns);
+    const informer = makeInformer(kc, `/apis/apps/v1/namespaces/${ns}/replicasets`, listFn);
+
+    informer.on('add', (_obj: V1ReplicaSet) => {
+      const previous = this.contextsState.get(context.name);
+      if (previous) {
+        previous.replicasetsCount++;
+      }
+      this.dispatchContextsState();
+    });
+
+    informer.on('delete', (_obj: V1ReplicaSet) => {
+      const previous = this.contextsState.get(context.name);
+      if (previous) {
+        previous.replicasetsCount--;
+      }
+      this.dispatchContextsState();
+    });
+    informer.on('error', (_err: unknown) => {
+      // Restart informer after 5sec
+      setTimeout(() => {
+        this.restartInformer(informer, context);
+      }, 5000);
+    });
+    this.restartInformer(informer, context);
+    return informer;
+  }
+
+  private restartInformer(informer: Informer<KubernetesObject> & ObjectCache<KubernetesObject>, context: KubeContext) {
+    const kubernetesConfiguration = this.configurationRegistry.getConfiguration('kubernetes');
+    const liveContextsInfo = kubernetesConfiguration.get<boolean>('LiveContextsInfo');
+    if (!liveContextsInfo) {
+      return;
+    }
+
+    informer
+      .start()
+      .then(() => {})
+      .catch((err: unknown) => {
+        console.log('==> catched err', err);
```

Comment:
There should be readable message that would explain where the error has happened.